### PR TITLE
Calendar Page

### PR DESCRIPTION
**Pull request by alderwhiteford**
Oh god this was gross. 

Things that need to be done:
* Migrate to event preview fetch
* tweak event preview fetch to return tags + host name
* clean up repetitive code in calendar comp

### CODE REVIEW

**Review by garrettladley — APPROVED** [review]
LGTM, good stuff @alderwhiteford

File: frontend/mobile/app/(design-system)/components/Calendar/Calendar.tsx [hunk 1]
```diff
@@ -1,106 +1,626 @@
...
+            ]);
+            old_events.current = [...allLoadedEvents];
+            return;
+        }
+
+        if (targetIndex === undefined) {
+            throw new Error('Target index must be defined');
+        }
+        if (type === 'left') {
+            setEvents((prevEvents) => {
```
Comment (alderwhiteford) [inline frontend/mobile/app/(design-system)/components/Calendar/Calendar.tsx:285]: pretty sure this code is the same, combine

File: frontend/mobile/app/(design-system)/components/Calendar/Calendar.tsx [hunk 2]
```diff
@@ -1,106 +1,626 @@
...
+            }
+        } else {
+            targetIndex = (mainIndex + 1) % 3;
+            data_index_array.current[targetIndex] -= 3;
+
+            // Update the current events in view:
+            updateEventsToCurrent(loadedEvents, 'right', targetIndex);
+
+            const distanceFromStart = data_index_array.current[targetIndex];
+            console.log(distanceFromStart);
```
Comment (alderwhiteford) [inline frontend/mobile/app/(design-system)/components/Calendar/Calendar.tsx:339]: remove console.log

File: frontend/mobile/app/(design-system)/components/Calendar/Calendar.tsx [hunk 3]
```diff
@@ -1,106 +1,626 @@
...
+    const handlePanAnimation = async (mainIndex: number) => {
+        const currIndex = data_index_array.current[mainIndex];
+        const beforeThreshold =
+            currIndex === refresh_threshold_item_count + 2 ||
+            currIndex ===
+                loadedEvents.length - refresh_threshold_item_count - 1;
+
+        // Card not been fully swiped:
+        if (
+            (record_translateX_array.current[mainIndex] < width / 3 &&
```
Comment (alderwhiteford) [inline frontend/mobile/app/(design-system)/components/Calendar/Calendar.tsx:357]: add comment here about this being where we can alter screen % swipe threshold

File: frontend/mobile/app/(design-system)/components/Calendar/Calendar.tsx [hunk 4]
```diff
@@ -1,106 +1,626 @@
...
+        nativeEvent
+    }: {
+        nativeEvent: Readonly<
+            GestureEventPayload & PanGestureHandlerEventPayload
+        >;
+    }) => {
+        // If the finger has been lifted
+        if (nativeEvent.state === State.END) {
+            old_translateX.current = 0;
+            handlePanAnimation(main_index.current);
```
Comment (alderwhiteford) [inline frontend/mobile/app/(design-system)/components/Calendar/Calendar.tsx:464]: curious to know if this is where we can avoid lag on data fetch swipe. Decouple from pan animation possibly?

File: frontend/mobile/app/(design-system)/components/Calendar/parser/calendarParser.ts [hunk 5]
```diff
@@ -0,0 +1,135 @@
+import { Event } from '@generatesac/lib';
+
+import { DAY_EPOCH_TIME } from '@/app/(design-system)/components/Calendar/Calendar';
+import {
+    EventPreview,
+    EventSection
+} from '@/app/(design-system)/components/Calendar/DayTimeSection';
+
+export const mockData: Event[] = [
```
Comment (alderwhiteford) [inline frontend/mobile/app/(design-system)/components/Calendar/parser/calendarParser.ts:9]: remove

**Pull request merged**